media/js/mozorg/home/takeover.js

```diff
@@ -0,0 +1,106 @@
+
+$(function () {
+    'use strict';
+
+    var $takeoverModal = $('#fundraising_takeover');
+    var $overlay = $('.page-overlay');
+    var $closeModal = $('#close_takeover');
+    var $form = $('#fundraiser-form');
+
+    var $donateOptionsSection = $('.donate-options');
+    var $donateCustom = $('#donate-custom');
+    var $optionLabels = $('label[class!="own-amount"]', $donateOptionsSection);
+    var $options = $('input[type="radio"]', $donateOptionsSection);
+
+    // sets the mozorg.takeover state to true indicating that the user has
+    // seen the takeover modal during their current session.
+    function storeSession() {
+        try {
+            sessionStorage.setItem('mozorg.takeover.seen', true);
+        } catch (ex) {
+            // yum, errors taste nice ;)
+        }
+    }
+
+    // Show and hides the takeover modal
+    // Removes the keyup event from body if specified
+    function toggleModal(removeListener) {
+
+        if(removeListener) {
+            // no longer listen for keyup events.
+            $('body').off('keyup.takeover');
+        }
+
+        $([$takeoverModal, $overlay]).each(function() {
+            $(this).toggleClass('hidden');
+        });
+    }
+
+    // removes checked from all the radio buttons
+    // and selected from the labels.
+    function clearOptions() {
+        $optionLabels.removeClass('selected');
+        $options.removeAttr('checked');
+    }
+
+    // handles clicks on the labels of the radio input elements.
+    $optionLabels.on('click', function() {
+        // clear selected class from labels
+        $optionLabels.removeClass('selected');
+        // empty any value that might have been typed
+        // into the custom amount field.
+        $donateCustom.val('');
+        // set the clicked label to selected
+        $(this).addClass('selected');
+    });
+
+    // once a user types into the custom amount field
+    // clear all other options.
+    $donateCustom.keyup(clearOptions);
+
+    // the name of the parameter that indicates the amount the user
+    // wish to donate has to be called amount. Seeing that we have
+    // radio buttons with that name as well as an input field, we need
+    // to disable the input field prior to form submission to avoid a
+    // scenario where we have ?amount=50&amount=
+    $form.on('submit', function() {
+        if ($donateCustom.val() === '') {
+            $donateCustom.attr('disabled', 'disabled');
+        }
+    });
+
+    // if a user clicks on 'continue to mozilla.org',
+    // close the modal and overlay
+    $closeModal.on('click', function(event) {
+        event.preventDefault();
+        toggleModal();
+    });
+
+    // modal is waffled so, first ensure it exists
+    if ($takeoverModal.length > 0) {
+        try {
+            // only show the takeover modal if the user has not already seen it
+            // during this session.
+            if (sessionStorage.getItem('mozorg.takeover.seen')) {
```

Comment:
This doesn't work, I don't see the takeover since the value is initially `null`. Suggest changing to `sessionStorage.getItem('mozorg.takeover.seen') !== 'true'` (make sure to set the value to `true` as a string above once visible).